app/src/main/java/com/android/sample/ui/listActivities/LikedActivities.kt

```diff
@@ -0,0 +1,254 @@
+
+  Scaffold(
+      modifier = modifier.testTag("likedActivitiesScreen"),
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { route -> navigationActions.navigateTo(route) },
+            tabList = LIST_TOP_LEVEL_DESTINATION,
+            selectedItem = navigationActions.currentRoute())
+      }) { paddingValues ->
+        Box(modifier = modifier.fillMaxSize().padding(paddingValues)) {
+          val likedActivitiesList by remember { mutableStateOf(profile?.likedActivities) }
+          when (uiState) {
+            is ListActivitiesViewModel.ActivitiesUiState.Success -> {
+              if (profile == null) {
+                Text(
+                    text = "You are not logged in, Login or Register to see your liked activities",
+                    modifier =
+                        Modifier.padding(8.dp)
+                            .align(Alignment.Center)
+                            .testTag("notConnectedPrompt"),
+                    color = MaterialTheme.colorScheme.onSurface)
+                Button(
+                    onClick = { navigationActions.navigateTo(Screen.SIGN_UP) },
+                    modifier = Modifier.testTag("signInButton")) {
+                      Text("Go to Sign In Page")
+                    }
+              }
+              if (likedActivitiesList != null) {
+                if (likedActivitiesList!!.isEmpty()) {
+                  Text(
+                      text = "There is no liked activity yet.",
+                      modifier =
+                          Modifier.padding(8.dp)
+                              .align(Alignment.Center)
+                              .testTag("emptyLikedActivityPrompt"),
+                      color = MaterialTheme.colorScheme.onSurface)
+                } else {
+
+                  LazyColumn(
+                      modifier =
+                          Modifier.padding(paddingValues)
+                              .fillMaxSize()
+                              .padding(16.dp)
+                              .padding(horizontal = 5.dp),
+                      verticalArrangement = Arrangement.spacedBy(16.dp)) {
+                        items(likedActivitiesList!!) { activityId ->
+                          ActivityCard2(
+                              activityId = activityId,
+                              navigationActions,
+                              viewModel,
+                              profileViewModel,
+                              profile,
+                              allActivities)
+                        }
+                      }
+                }
+              }
+            }
+            is ListActivitiesViewModel.ActivitiesUiState.Error -> {
+              val error = (uiState as ListActivitiesViewModel.ActivitiesUiState.Error).exception
+              Text(text = "Error: ${error.message}", modifier = Modifier.padding(8.dp))
```

Comment:
The implementation provides clear user feedback for different scenarios—whether the user is not logged in or there are no liked activities.